private/define-expanders.rkt

```diff
@@ -7,21 +7,32 @@
 
 (provide define-expander-type)
 
+(define-for-syntax (remove-use-site-scope stx)
+  (define bd
+    (syntax-local-identifier-as-binding (syntax-local-introduce #'here)))
+  (define delta
+    (make-syntax-delta-introducer (syntax-local-introduce #'here) bd))
+  (delta stx 'remove))
+
 (define-syntax define-expander-type
   (syntax-parser
     [(_ name:id)
      (with-derived-ids #'name ([?-expander-type "~a-expander-type"]
                                [make-?-expander "make-~a-expander"]
                                [?-expander? "~a-expander?"]
                                [define-?-expander "define-~a-expander"]
+                               [define-?-expander-bug "define-~a-expander-bug"]
                                [expand-all-?-expanders "expand-all-~a-expanders"])
-                       #'(begin
+                       #`(begin
                            (define-for-syntax ?-expander-type (make-expander-type))
                            (define-for-syntax (make-?-expander transformer)
                              (expander ?-expander-type transformer))
                            (define-for-syntax (?-expander? v)
                              (expander-of-type? ?-expander-type v))
-                           (define-syntax-rule (define-?-expander expander-name transformer)
-                             (define-syntax expander-name (make-?-expander transformer)))
+                           (define-syntax (define-?-expander stx)
+                             (syntax-case stx ()
+                               [(_ expander-name transformer)
```

Comment:
Using syntax-parse and checking that `expander-name` is an identifier and `transformer` an expression would be a good improvement to add here while you're at it.